launcher/meta/Index.cpp

```diff
@@ -159,4 +159,79 @@ Version::Ptr Index::getLoadedVersion(const QString& uid, const QString& version)
     ev.exec();
     return get(uid, version);
 }
+
+// https://github.com/PrismLauncher/PrismLauncher/issues/3866
+bool isWeirdForgeFormat(const QString& input)
+{
+    int dashIndex = input.indexOf('-');
+    if (dashIndex == -1 || dashIndex + 1 >= input.length())
+        return false;  // No dash or nothing after dash - considered valid
+
+    QChar nextChar = input[dashIndex + 1];
+    if (nextChar.isLetter()) {
+        return false;  // Immediate letter after dash
+    }
+
+    QString afterDash = input.mid(dashIndex + 1);
+    return afterDash.contains('.');  // Dot found after dash so most probably it is
+}
```

Comment:
This handling should be ATLauncher-specific as no other format is likely to have it

Reply:
The artifact is present in the forge Maven repo: https://mvnrepository.com/artifact/net.minecraftforge/forge/1.12.2-14.23.5.2847 (one example)
So I will presume that the other formats can still have it

Reply:
I edited message
> Since you're just using this to obtain the Minecraft version can't you just use mcVersion?

Reply:
I'm using it to find the forge version

Reply:
Still not really sure about it as ideally our meta handling code should not be intertwined with loader specific stuff...

Reply:
I can remove this version patch from here, and it should use the recommended version instead of the intended version. Are you okay with that?

Reply:
Done removed that weird handling: now instead of searching a specific version that may not be there, it will always use the recommended version if the format return a loader that is not in our meta